java-frontend/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -139,30 +140,17 @@ private static Type best(List<Type> minimalCandidates) {
   }
 
   private static List<Set<Type>> supertypes(Iterable<Type> types) {
-    try {
-      List<Set<Type>> results = new LinkedList<>();
-      for (Type type : types) {
-        checkParametrizedType(type);
-        Set<Type> supertypes = new LinkedHashSet<>();
-        supertypes.add(type);
-        for (Type supertype : ((JavaType) type).symbol.superTypes()) {
-          checkParametrizedType(supertype);
-          supertypes.add(supertype);
-        }
-        results.add(supertypes);
+    List<Set<Type>> results = new LinkedList<>();
+    for (Type type : types) {
+      Set<Type> supertypes = new LinkedHashSet<>();
+      supertypes.add(type.erasure());
+      for (Type supertype : ((JavaType) type.erasure()).symbol.superTypes()) {
```

Comment:
This line needs a comment to highlight the fact that we rely currently on erasure() (even if already present in unit test)